src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy

```diff
@@ -637,13 +667,50 @@ public abstract class GenerateProtoTask extends DefaultTask {
 
   protected String computeExecutablePath(ExecutableLocator locator) {
     if (locator.path != null) {
-      return locator.path
+      return locator.path.endsWith(JAR_SUFFIX) ? createJarTrampolineScript(locator.path) : locator.path
     }
     File file = locatorToAlternativePathsMapping.getting(locator.name).get().singleFile
+    if (file.name.endsWith(JAR_SUFFIX)) {
+      return createJarTrampolineScript(file.getAbsolutePath())
+    }
+
     if (!file.canExecute() && !file.setExecutable(true)) {
       throw new GradleException("Cannot set ${file} as executable")
     }
     logger.info("Resolved artifact: ${file}")
     return file.path
   }
+
+  /**
+   * protoc only supports plugins that are a single self contained executable file. For .jar files create a trampoline
+   * script to execute the jar file. Assume the jar is a "fat jar" or "uber jar" and don't attempt any artifact
+   * resolution.
+   * @param jarAbsolutePath Absolute path to the .jar file.
+   * @return The absolute path to the trampoline executable script.
+   */
+  private String createJarTrampolineScript(String jarAbsolutePath) {
+    assert jarAbsolutePath.endsWith(JAR_SUFFIX)
+    boolean isWindows = isWindows()
+    File scriptExecutableFile = new File("${project.buildDir}/scripts/" +
+            jarAbsolutePath[0..(jarAbsolutePath.length() - JAR_SUFFIX.length())] + "-trampoline." +
```

Comment:
Does this create the original folder structure under scripts/? How does that work on Windows with it's `C:`?

If I'm reading this right, I'm worried about how long this filename will be. We may hit max path limits. I suggest we use the task name to make the script name unique, and we can still use the jar file name to make the command line informative.

Reply:
> Does this create the original folder structure under scripts/? 

good catch. this isn't necessary and just using the jar file name (without extension) should be sufficient. Can you clarify what you mean by "task name" (e.g. gradle task)? The jar file name includes the version information for uniqueness and is easy to correlate with the jar file it executes.

```
file name: ${buildDir}/scripts/servicetalk-grpc-protoc-0.31.0-SNAPSHOT-all-trampoline.sh

execute jar: <some-path>/servicetalk-grpc-protoc-0.31.0-SNAPSHOT-all.jar
```

Reply:
I mean the GenerateProtoTask instance's name. Every `Task` in gradle [has a unique name](https://docs.gradle.org/current/javadoc/org/gradle/api/Task.html#getName--) within the gradle project, so we can use it to prevent conflicts.

Even assuming the jar file name is unique (which it isn't a guarantee; there's no guarantee this file is coming from a well-known central hosting with nice names), we'd still need to unique-ify it in case two GenerateProtoTasks run at the same time using the same jar.

Reply:
done